### ISSUE TITLE
tt-blacksmith CI

### ISSUE THREAD

**Issue by ndrakulicTT** [tenstorrent/tt-blacksmith#308]
Current status is that there exists gh action `Run Test` introduced in #259. But it has some falling tests.
### On Push
- Run only mnist tests in every combination signlechip/tp/dp/... and torch/jax variant.
### On nightly - tt-xla uplift/
- Run most of the examples that we have in the repo

| Framework | Model         | Dataset     | Method                      | Devices | Details                                                                                                                | Done    |
| --------- | ------------- | ----------- | --------------------------- | ------- | ---------------------------------------------------------------------------------------------------------------------- | ------- |
| PyTorch   | MLP           | MNIST       | Full-model                  | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/mnist/README.md)          | ✅      |
| PyTorch   | MLP           | MNIST       | Full-model, Data parallel   | N300    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/mnist/README.md)          | ✅      |
| PyTorch   | MLP           | MNIST       | Full-model, Tensor parallel | N300    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/mnist/README.md)          | ✅      |
| PyTorch   | Llama 3.2 1B  | SST-2       | LoRA                        | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/llama/xla/lora/README.md) | ⏩       |
| PyTorch   | Llama 3.2 1B  | SST-2       | LoRA, Data parallel         | T3K     | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/llama/xla/lora/README.md) | ⏩       |
| PyTorch   | Qwen 2.5 1.5B | Text-to-SQL | LoRA                        | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/qwen/README.md)           | ⏩       |
| PyTorch   | Gemma 1.1 2B  | Squad-V2    | LoRA                        | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/gemma11/README.md)        | ⏩       |
| PyTorch   | ALBERT        | Banking77   | Adapters                    | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/albert/README.md)         | ⏩       |
| PyTorch   | Phi-1         | SST-2       | LoRA                        | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/torch/phi/README.md)            | ⏩      |
| JAX       | MLP           | MNIST       | Full-model                  | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/jax/mnist/README.md)            | ✅      |
| JAX       | MLP           | MNIST       | Full-model, Data parallel   | N300    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/jax/mnist/README.md)            | ✅      |
| JAX       | MLP           | MNIST       | Full-model, Tensor parallel | N300    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/jax/mnist/README.md)            | ✅      |
| JAX       | NeRF          | Blender     | Full-model                  | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/jax/nerf/README.md)             | ❌      |
| JAX       | Llama 3.2 1B  | SST-2       | DoRA                        | N150    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/jax/llama_dora/README.md)       | ❌      |
| JAX       | DistilBERT    | SST-2       | Distillation, Data parallel | N300    | [README](https://github.com/tenstorrent/tt-blacksmith/blob/main/blacksmith/experiments/jax/distil_bert/README.md)      | ❌      |

### For each test
- Have pre-saved checkpoints from the reference run
- Load the random or preselected checkpoint
- Train the model in duration of ~10% of full run or up to 1h
	- This will need custom yaml configuration for each experiment
- Check the loss and accuracy curves against referent values
- Check if the tests have saved all expected artifacts
